src/mono/msbuild/AndroidTestRunner/AndroidTestRunner.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.Linq;
+using System.Text;
+using System.IO;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+using Microsoft.DotNet.XHarness.Tests.Runners;
+using Microsoft.DotNet.XHarness.Tests.Runners.Core;
+
+public class SimpleAndroidTestRunner : AndroidApplicationEntryPoint, IDevice
+{
+    private static List<string> s_testLibs = new List<string>();
+    private static string? s_MainTestName;
+
+    public static async Task<int> Main(string[] args)
+    {
+        s_testLibs = Directory.GetFiles(Environment.CurrentDirectory, "*.Tests.dll").ToList();
+        if (s_testLibs.Count < 1)
+        {
+            Console.WriteLine($"Test libs were not found (*.Tests.dll was not found in {Environment.CurrentDirectory})");
+            return -1;
+        }
+        s_MainTestName = Path.GetFileNameWithoutExtension(s_testLibs[0]);
+        var simpleTestRunner = new SimpleAndroidTestRunner(true);
+        await simpleTestRunner.RunAsync();
+        Console.WriteLine("----- Done -----");
+        return 0;
+    }
+
+    public SimpleAndroidTestRunner(bool verbose)
+    {
+        if (verbose)
+        {
+            MinimumLogLevel = MinimumLogLevel.Verbose;
+            _maxParallelThreads = 1;
+        }
+        else
+        {
+            MinimumLogLevel = MinimumLogLevel.Info;
+            _maxParallelThreads = Environment.ProcessorCount;
+        }
+    }
+
+    protected override IEnumerable<TestAssemblyInfo> GetTestAssemblies()
+    {
+        foreach (string file in s_testLibs)
+        {
+            yield return new TestAssemblyInfo(Assembly.LoadFrom(file), file);
+        }
+    }
+
+    protected override void TerminateWithSuccess()
+    {
+        Console.WriteLine("[TerminateWithSuccess]");
+    }
+
+    private int? _maxParallelThreads;
+
+    protected override int? MaxParallelThreads => _maxParallelThreads;
+
+    protected override IDevice Device => this;
+
+    protected override TestRunnerType TestRunner => TestRunnerType.Xunit;
+
+    protected override string? IgnoreFilesDirectory => Environment.CurrentDirectory;
+
+    public string BundleIdentifier => "net.dot." + s_MainTestName;
+
+    public string? UniqueIdentifier { get; }
+
+    public string? Name { get; }
+
+    public string? Model { get; }
+
+    public string? SystemName { get; }
+
+    public string? SystemVersion { get; }
+
+    public string? Locale { get; }
+
+    public override TextWriter? Logger => null;
+
+    public override string TestsResultsFinalPath => 
+        Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Personal), "Logs.txt");
```

Comment:
isn't this the xunit xml results path? we should make it something like `testResults.xml`

Reply:
no idea why it forces me to specify it, iOS doesn't not. will ask the xharness team

Reply:
I think it's fine to specify it (xharness ought to grab this file from the device), but we should call it `testResults.xml` not `Logs.txt`

Reply:
I'll change, it was a draft since I am waiting for iOS fixes to land first before I wire this one